tdp/cli/commands/plan/edit.py

```diff
@@ -0,0 +1,214 @@
+            raise ValueError(
+                "Error on line '" + line + "' : host is required after 'on' keyword."
+            )
+    else:
+        host = None
+
+    # Get the extra vars and test if they are declared
+    if parsed_line.group(3) is not None:
+        extra_vars = parsed_line.group(3).split(" with ")[1]
+        if extra_vars == "":
+            raise ValueError("Extra vars are required after 'with' keyword.")
+        extra_vars = extra_vars.split(",")
+        extra_vars = [item.strip() for item in extra_vars]
+    else:
+        extra_vars = None
+
+    return (operation, host, extra_vars)
+
+
+@click.command(short_help="Edit Deployment plan.")
+@collections
+@database_dsn
+def edit(
+    collections,
+    database_dsn,
+):
+    with get_session(database_dsn, commit_on_exit=True) as session:
+        planned_deployment_log = get_planned_deployment_log(session)
+        if planned_deployment_log is None:
+            raise click.ClickException(
+                "No planned deployment found, please run `tdp plan` first."
+            )
+
+        operation_list = [
+            (operation.operation, operation.host, operation.extra_vars)
+            for operation in planned_deployment_log.operations
+        ]
+
+        operation_lines = [
+            operation
+            + (" on " + host if host else "")
+            + (" with " + ",".join(extra_vars) if extra_vars else "")
+            for operation, host, extra_vars in operation_list
+        ]
+
+        temp_file = tempfile.NamedTemporaryFile(mode="w", suffix=".csv", delete=False)
+
+        header_text = f"""\
+# ------------------- DEPLOYMENT PLAN EDITOR --------------------------
+# Modify this file to edit the deployment plan.
+# Lines beginning with '#' are comments and will be ignored.
+#
+# Current plan id: {planned_deployment_log.id}
+# Temporary file: {temp_file.name}
+#
+# Execution Order:
+# - Operations are executed in the order they appear below.
+# - You can add, remove, or rearrange operations.
+# - Use `tdp nodes` command to view a list of available operations.
+#
+# Operations options:
+# - Specify a host with the `on` keyword.
+#   Example: hdfs_datanode_config on worker_02
+# - Add extra variables using the `with` keyword and separate with `,`.
+#   Example: wait_sleep with rolling-interval=5, mock-arg=example
+# ------------------- DEPLOYMENT PLAN EDITOR --------------------------
+"""
+
+        temp_file.write(header_text + "\n".join(operation_lines))
+        temp_file.close()
+
+        click.echo(
+            f"Current plan id: {planned_deployment_log.id}\nTemporary file : {temp_file.name}"
+        )
+
+        try:
+            program_exit = False
+            # Loop until plan has no error or if user wants to stop editing the plan
+            while not program_exit:
+                # Loop until user wants to save editing the plan
+                program_continue = False
+                while not program_continue:
```

Comment:
I really think that it would be better to use a context manager along with a `while True:`/`break`/`continue` statements. It would avoid all this nesting.

Reply:
Done, I used what we did in #415